bin/update_ati_analytics.py

```diff
@@ -0,0 +1,125 @@
+"""
+Gets the ati-informal-requests-analytics.csv from the SMB
+and filters out the current month's rows and then patches the
+Resource on the Registry.
+"""
+
+import os
+import click
+import csv
+import traceback
+import tempfile
+from datetime import datetime
+import subprocess
+from six import PY2
+from shutil import rmtree
+from codecs import BOM_UTF8
+
+if PY2:
+    from cStringIO import StringIO
+else:
+    from io import StringIO
+
+
+def error_message(message):
+    click.echo("\n\033[1;33m%s\033[0;0m\n\n" % message)
+
+
+def success_message(message):
+    click.echo("\n\033[0;36m\033[1m%s\033[0;0m\n\n" % message)
+
+
+def get_smb_file():
+    """
+    Get the file path generated from Drupal
+    The SMB directories are different on some servers
+    hence the checks for the different file paths
+    """
+    check_paths = [
+        '/opt/tbs/ckan/smb_portal/portal/public/ati-informal-requests-analytics.csv',
+        '/opt/tbs/ckan/smb/portal/public/ati-informal-requests-analytics.csv',
+        '/opt/tbs/ckan/smb_portal/portal/public/static/ati-informal-requests-analytics.csv',
+        '/opt/tbs/ckan/smb/portal/public/static/ati-informal-requests-analytics.csv',]
+    for file in check_paths:
+        if os.path.isfile(file):
+            return file
+    error_message('/opt/tbs/ckan/[smb_portal|smb]/portal/public/[static/]ati-informal-requests-analytics.csv does not exist.')
+
+
+@click.command(short_help="Filters ATI Informal Analytics CSV and uploads to Registry Resource.")
+@click.option('-c', '--config', required=True, envvar='REGISTRY_INI', type=click.File('r'),
+              help='CKAN config INI file. Defaults to the $REGISTRY_INI environment variable.')
+@click.option('-r', '--resource-id', required=False, type=click.STRING, default='e664cf3d-6cb7-4aaa-adfa-e459c2552e3e',
+              help='Resource ID to patch the file to. Defaults to e664cf3d-6cb7-4aaa-adfa-e459c2552e3e.')
+@click.option('-v', '--verbose', is_flag=True, type=click.BOOL, help='Increase verbosity.')
+def update(config, resource_id, verbose):
+    """
+    Filters ATI Informal Analytics CSV and uploads to Registry Resource.
+    """
+    generated_file = get_smb_file()
+    if not generated_file:
+        return
+    if PY2:
+        temp_dir = tempfile.mkdtemp()
+    else:
+        temp_dir = tempfile.TemporaryDirectory()
+    temp_file = os.path.join(temp_dir, 'ati-informal-requests-analytics.csv')
+
+    today = datetime.today()
+
+    errors = StringIO()
+    skipped_rows = 0
+    command = 'ckanapi action resource_patch -c %s id=%s upload@"%s"' \
```

Comment:
better to use a list so that subprocess.call can use execv and shell=False to skip shell parsing of the string. These parameters may not cause a problem but it's a good habit regardless.

Reply:
Okay using a list now without `shell=True`, it is just failing on the `ckanapi` command usage so will need to look into that

Reply:
@wardi okay that should be fixed now. It was because I had the extra `ckanapi` in the command and I guess that is only needed for the python2 subprocess and not the python3 one.